### PR TITLE
add method to convert imperative questions to interrogative questions

### PR DESCRIPTION
**Pull request by nelson-liu**
I wrote a function that converts questions typically found in science exams (e.g. "Explain this", "Describe that", "Name these") to questions that are more squad-like. The tests might be good examples for what behavior it does specifically --- I also tried to comment inline heavily to reduce confusion later down the line as the regex can be pretty inscrutable.

### CODE REVIEW

File: src/main/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReader.scala [hunk 1]
```diff
@@ -29,4 +30,47 @@ class OmnibusDaDatasetReader extends DatasetReader[DirectAnswerInstance] {
...
+
+      // If there is an imperative, remove all words that occur before
+      // it (the first imperative) in the sentence. Thus, "Please briefly explain this." becomes
+      // "explain this."
+      imperativeModifiers = s"""(?i).+?(?=(${imperatives}) )""".r
+      imperativeModifierRemovedSentence = imperativeModifiers.replaceFirstIn(imperativeSentence, "")
+    } yield imperativeModifierRemovedSentence
+    // For each sentence, the imperative words will be replaced with "What is" if
+    // they occur at the start of the sentence. However, if these
+    // imperative words occur directly before the the interrogative words, we just
```
Comment (matt-gardner) [inline src/main/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReader.scala]: `s/the the/the/`

File: src/test/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReaderSpec.scala [hunk 2]
```diff
@@ -21,11 +23,54 @@ class OmnibusDaDatasetReaderSpec extends FlatSpecLike with Matchers {
   val questionText2 = "Coal is a nonrenewable energy resource. People burn coal to " +
     "make electricity and heat buildings. Explain how long it would most likely take for new coal " +
     "to form."
+  val transformedQuestionText2 = "Coal is a nonrenewable energy resource. People burn coal to " +
+    "make electricity and heat buildings. How long it would most likely take for new coal to form."
```
Comment (matt-gardner) [inline src/test/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReaderSpec.scala:27]: This is getting increasingly brittle, but you could consider a simple regex to replace the order of "would" and "it" here.  I'd want to see more examples of this before deciding on whether this was a good idea and how to do it, though.

File: src/test/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReaderSpec.scala [hunk 3]
```diff
@@ -21,11 +23,54 @@ class OmnibusDaDatasetReaderSpec extends FlatSpecLike with Matchers {
...
+
+  val questionText4 = "Even though an adult amphibian can live on land, why must it " +
+    "return to the water? (Explain.)"
+  val transformedQuestionText4 = "Even though an adult amphibian can live on land, " +
+    "why must it return to the water? (Explain.)"
+
+  // TODO(nelson): Would like to be able to handle plural/singular.
+  val questionText5 = "Name four organisms you might find living in or around a " +
+    "freshwater lake."
+  val transformedQuestionText5 = "What is four organisms you might find living in or around a " +
```
Comment (matt-gardner) [inline src/test/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReaderSpec.scala:46]: I would ask Sumithra what she did about questions like this that ask for a specific number of things.

File: src/test/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReaderSpec.scala [hunk 4]
```diff
@@ -21,11 +23,54 @@ class OmnibusDaDatasetReaderSpec extends FlatSpecLike with Matchers {
...
+  "freshwater lake."
+  val questionText6 = "List the conditions that must be present in order for a " +
+  "seed to germinate."
+  val transformedQuestionText6 = "What is the conditions that must be present in "+
+  "order for a seed to germinate."
+
+  // TODO(nelson): not clear how to insert the proper verbs here.
```
Comment (matt-gardner) [inline src/test/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReaderSpec.scala:53]: You want to rotate the dependencies in the dependency tree, moving the subject "most animal life on Earth" after the auxiliary verb "could not".  But being able to do that correctly assumes a correct dependency parse, which we don't have.  So we should just punt on this, unless you want to write regexes over POS tags...

**Review by matt-gardner — APPROVED** [review]
Good enough for a first pass at this.

File: src/test/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReaderSpec.scala (hunk 4 above)
Comment (nelson-liu) [reply src/test/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReaderSpec.scala:53]: yeah, i think i'll pass for now...

File: src/test/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReaderSpec.scala (hunk 2 above)
Comment (nelson-liu) [reply src/test/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReaderSpec.scala:27]: i've put it on the todo trello board.

**Pull request merged**